addons/web_editor/static/src/js/editor/odoo-editor/test/spec/editor.test.js

```diff
@@ -2297,6 +2353,101 @@ X[]
+                        contentBefore: unformat(`
+                            <div contenteditable="false">
+                                <div contenteditable="true">
+                                    <p>[]content</p>
+                                </div>
+                            </div>
+                        `),
+                        stepFunction: deleteBackward,
+                        contentAfter: unformat(`
+                            <div contenteditable="false">
+                                <div contenteditable="true">
+                                    <p>[]content</p>
+                                </div>
+                            </div>
+                        `),
+                    });
+                });
+                it('should move a line before the uneditable nesting zone if the cursor is at the start of the first element of the nested zone', async () => {
+                    await testEditor(BasicEditor, {
+                        contentBefore: unformat(`
+                            <div contenteditable="false">
+                                <div contenteditable="true">
+                                    <p>[]content</p>
+                                    <p>othercontent</p>
+                                </div>
+                            </div>
+                        `),
+                        stepFunction: deleteBackward,
+                        contentAfter: unformat(`
+                            <p>[]content</p>
+                            <div contenteditable="false">
+                                <div contenteditable="true">
+                                    <p>othercontent</p>
+                                </div>
+                            </div>
+                        `),
+                    });
+                });
+                it('should merge with a line before the uneditable nesting zone if the cursor is at the start of the first element of the nested zone', async () => {
+                    await testEditor(BasicEditor, {
+                        contentBefore: unformat(`
+                            <p>previouscontent</p>
+                            <div contenteditable="false">
+                                <div contenteditable="true">
+                                    <p>[]content</p>
+                                    <p>othercontent</p>
+                                </div>
+                            </div>
+                        `),
+                        stepFunction: deleteBackward,
+                        contentAfter: unformat(`
+                            <p>previouscontent[]content</p>
+                            <div contenteditable="false">
+                                <div contenteditable="true">
+                                    <p>othercontent</p>
+                                </div>
+                            </div>
+                        `),
+                    });
+                });
```

Comment:
In fact I feel like this kind of behaviors have the potential to break a few things in website, since it allows some content to be displaced outside of the non-editable zone boundaries, which might be unexpected in website, though I might be wrong on that.

EDIT: See the EDIT in the previous comment where I suggest to add the concept of permeability.